envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/envoy/EnvoyAdmin.kt

```diff
@@ -70,6 +71,7 @@ class EnvoyAdmin(
         val configDump = configDump()
         val bootstrapConfigDump = bootstrapConfigDump(configDump)
         val node = bootstrapConfigDump.at("/bootstrap/node")
+        (node as ObjectNode).remove("hidden_envoy_deprecated_build_version")
```

Comment:
Otherwise it won't decode.

Reply:
If I see correctly we only *encode* `node` to a json string. No field should be a problem. What do you mean by "decode additional fields"?

Reply:
I'm getting 

```{"timestamp":"2020-03-04T18:13:51.101+0000","status":400,"error":"Bad Request","message":"I/O error while reading input message; nested exception is com.google.protobuf.InvalidProtocolBufferException: Cannot find field: hidden_envoy_deprecated_build_version in message envoy.api.v2.core.Node","path":"/snapshot"}``` 

When trying to POST to `/snapshot`. This is because `/config_dump`  has `hidden_envoy_deprecated_build_version` field  (probably comming from: https://github.com/envoyproxy/envoy/blame/c5c1e5bb6570f81f30dd48d8b379392c84e81fe0/generated_api_shadow/envoy/config/core/v3/base.proto ) but it's not in java-control-plane `Node` proto.